app/serializers/basic_post_serializer.rb

```diff
@@ -38,7 +38,27 @@ def cooked
         I18n.t('flagging.user_must_edit')
       end
     else
-      object.filter_quotes(@parent_post)
+      cooked = object.filter_quotes(@parent_post)
+
+      if scope&.user
+        group = Group
```

Comment:
If you have the `id` of the group, then this could become

```ruby
group_id = object.custom_fields["membership_request"]

if scope&.user && GroupUser.where(user: scope.user, group_id: group_id, owner: true).exists?
  group = Group.find(group_id)
  cooked << <<~HTML
    <hr />
    <a href="#{Discourse.base_uri}/g/#{group.name}/requests">
      #{I18n.t('groups.request_membership_pm.handle')}
    </a>
  HTML
end
```